tensorflow/python/ops/array_ops.py

```diff
@@ -3628,7 +3645,114 @@ def _get_paddings_constant(paddings):
     return [_get_paddings_constant(x) for x in paddings]
   else:
     return paddings
+  
+def _wrap_pad(tensor, paddings, name=None):
+  """Helper to create wrap pad option with ops already built.
 
+  See `pad` for more info on how this function works.
+  """
+  # Use constant padding to create a tensor with final shape and original values.
+  padded = gen_array_ops.pad(input=tensor, paddings=paddings, name=name)
+  start_coords, end_coords = [
+    shape(padded) - paddings[:,1] - shape(tensor), 
+    shape(padded) - paddings[:,1]
+    ]
+  tensor_to_copy = tensor
+  strides = ones(rank(padded))
+  # Add wraps for each dimension.
+  for i in range(shape(paddings)[0]):
+    # The shape of `tensor_to_copy` can change in iterations. 
+    # Set the loop options to avoid an error in graph 
+    # mode.
+    directives.set_loop_options(
+                shape_invariants=[(tensor_to_copy, 
+                                   tensor_shape.TensorShape(None))]
+            )
+    lower_idx, upper_idx = paddings[i][0], paddings[i][1]
+    lower_begin = identity(start_coords)
+    upper_begin = identity(start_coords)
+    # Add the lower wrap.
+    while lower_idx > 0:
+      # Check for partial wrap.
+      if lower_idx < shape(tensor)[i]:
+        partial_slice_begin = zeros(rank(tensor_to_copy), dtype=dtypes.int32)
+        partial_slice_begin = gen_array_ops.tensor_scatter_nd_add(
```

Comment:
Hmm... apparently this doesn't exist.  I'm seeing:
```
Traceback (most recent call last):
  File "tensorflow/python/ops/array_ops_test.py", line 157, in testWrapPadOp
    array_ops._wrap_pad(t, paddings),
  File "tensorflow/python/ops/array_ops.py", line 3677, in _wrap_pad
    partial_slice_begin = gen_array_ops.tensor_scatter_nd_add(
AttributeError: module 'tensorflow.python.ops.gen_array_ops' has no attribute 'tensor_scatter_nd_add'
```

Reply:
I think the generated op is just `tensor_scatter_add` (minus the nd).

Reply:
updated this instance + another couple of these that creeped in. 

(sorry been struggling to run tests locally so didn't check)